spock-core/src/main/java/org/spockframework/mock/runtime/IMockMaker.java

```diff
@@ -0,0 +1,317 @@
+     */
+    ClassLoader getClassLoader();
+
+    /**
+     * @return whether class-based mock objects should be instantiated with the Objenesis library (if available),
+     * or by calling a real constructor, if the {@link IMockMaker} supports the Objenesis library.
+     */
+    boolean isUseObjenesis();
+
+    /**
+     * @return {@code true} if this mock shall mock the static methods of the {@link #getMockType()}.
+     */
+    boolean isStaticMock();
+
+    /**
+     * @return the {@link IMockMakerSettings} passed from the user containing the {@link IMockMakerSettings#getCustomData()}, or {@code null}.
+     */
+    @Nullable
+    IMockMakerSettings getMockMakerSettings();
+  }
+
+  /**
+   * Custom settings passed to the {@link IMockMaker} and created for explicitly for the requested mock maker.
+   * The user will create such an instance at the mocking site: {@code Mock(mockMaker:yourMockSettings{yourProperty = true}, ClassToMock)}
+   * <p>
+   * The {@link IMockMaker} shall define a static constant or method (see {@link MockMakers}), which let the user construct custom {@link IMockMakerSettings}.
+   * <p>
+   * If you provide a method, you can define your settings in a typesafe manner for the user, e.g. with a {@link groovy.lang.Closure} parameter to configure the mock.
+   *
+   * @since 2.4
+   */
+  @Immutable
+  @Beta
+  interface IMockMakerSettings {
+
+    /**
+     * Creates a simple {@link IMockMakerSettings} object with only an ID.
+     *
+     * @param id the mock maker ID.
+     * @return the settings object
+     */
+    static IMockMakerSettings simple(String id) {
+      return new IMockMakerSettings() {
+        @Override
+        public String getMockMakerId() {
+          return id;
+        }
+
+        @Override
+        public String toString() {
+          return id;
+        }
+      };
+    }
+
+    String getMockMakerId();
+  }
+
+  /**
+   * {@link MockMakerCapability} describes the possible features of an {@link IMockMaker}.
+   *
+   * <p>This provides the ability to extend the list of capabilities in the future without breaking existing external {@link IMockMaker}s.</p>
+   *
+   * @since 2.4
+   */
+  @Beta
+  enum MockMakerCapability {
+    /**
+     * Is able to mock an interface type.
+     */
+    INTERFACE("Cannot mock interfaces."),
+    /**
+     * Is able to mock a class type.
+     */
+    CLASS("Cannot mock classes."),
+    /**
+     * Is able to mock a class/interface type with additional interfaces.
+     */
+    ADDITIONAL_INTERFACES("Cannot mock classes with additional interfaces."),
+    /**
+     * Is able to pass explicit constructor arguments to the class constructor.
+     */
+    EXPLICIT_CONSTRUCTOR_ARGUMENTS("Explicit constructor arguments are not supported."),
+    /**
+     * Is able to mock a final class type.
+     */
+    FINAL_CLASS("Cannot mock final classes."),
```

Comment:
I am not yet sure with sealed classes, because we can't ask that with Java 8 API.
I would add that in the future, when we know how to ask the runtime, if a class is sealed, because then we can add the global feature check to the `MockMakerRegistry`.
I would investigate that in a new PR.

Reply:
Sealed classes can be check using `Class#isSealed()`, so we will probably either need to have some Java 17 specific code, or use the reflection reflectively (:-D), or use Groovy to do that check, if the sealed classes behave like final classes here, that was my main question.
But yeah, checking it in a separate PR sounds good too. :-)